src/syntax/token.rs

```diff
@@ -500,7 +503,9 @@ impl<'c> Iterator for Tokenizer<'c> {
         match self.curr.unwrap() {
             // comment
             '#' => Some(self.next_comment()),
-            // integer
+            // negative integer
+            '-' if self.next.unwrap().is_digit(10) => Some(self.next_int()),
```

Comment:
Don't use `unwrap()` here; `self.next` may be None (indicating EOF), and we haven't done a check to see if it is, or not. If the last character of a file is `-` then it will cause the compiler to panic. Use `self.next.map(|c| c.is_digit(10)).unwrap_or(false)` to avoid a panic.